bangazonapi/views/product.py

```diff
@@ -305,3 +305,47 @@ def recommend(self, request, pk=None):
             return Response(None, status=status.HTTP_204_NO_CONTENT)
 
         return Response(None, status=status.HTTP_405_METHOD_NOT_ALLOWED)
+
+    @action(methods=['post', 'delete'], detail=True)
+    def like(self, request, pk=None):
```

Comment:
Right now, there's nothing to prevent duplicate `Like` objects from being added to the database when a user sends a `POST` request to `products/x/like`. After `liking` the same product several times, I tried `unliking` it, which resulted in an error.

```
    "message": "get() returned more than one Like -- it returned 13!"
```
You'll need to modify your `like` action to prevent users from liking the same product more than once.

Reply:
Edited like action to catch multiple instances of identical like. Server now returns 400 BAD REQUEST error and message detailing that an identical like already exists in db.